### PR TITLE
Allow WRITE or REACT_REPLY for channel message writes from stream node

### PR DESCRIPTION
**Pull request by clemire**
Follow-up work is to disallow unpermitted messages from the client on send. Once this is done, the backend of react-reply is basically finished.

### CODE REVIEW

**Review by texuf — COMMENTED** [review]
Can you please make the api change in its own PR? Thank you.

**Comment by clemire** [comment]
> Can you please make the api change in its own PR? Thank you.

No problem, I'll split the react-reply stuff out.

**Comment by clemire** [comment]
> > Can you please make the api change in its own PR? Thank you.
> 
> No problem, I'll split the react-reply stuff out.

FYI https://github.com/river-build/river/pull/361

File: packages/sdk/src/channelsWithEntitlements.test.ts [hunk 1]
```diff
@@ -190,6 +190,119 @@ async function expectUserCannotJoinChannel(
...
+                    body: 'Hello, world!',
+                    mentions: [],
+                    attachments: [],
+                },
+                threadId: refEventId, // reply to Bob's message
+            }),
+        ).toResolve()
+
+        // Top-level post currently allowed.
+        // TODO: after client is updated to reject unpermitted self-posts, this should reject.
```
[the remark this answers is not in the document]
Comment (clemire) [reply packages/sdk/src/channelsWithEntitlements.test.ts:258]: The client will filter itself while sending, so we can introspect into the payload before it's encrypted. We scoped new payload types out for this work.

[the remark this answers is not in the document]
Comment (brianathere) [reply packages/sdk/src/channelsWithEntitlements.test.ts:258]: @erikolsson @texuf To start with we're adding the permission and in the sending client we will not send if the user doesn't have it. Then, we will later make a protocol change to hoist the type of event out of the encrypted portion of the payload so we can also check the permission in the node. This allows us to surface the feature and get it into clients, and then later add enforcement in the node.

**Review by brianathere — APPROVED** [review]

**Pull request merged**